controllers/contentlibrary/v1alpha2/contentlibraryitem/contentlibraryitem_controller.go

```diff
@@ -245,7 +249,32 @@ func (r *Reconciler) setUpVMIFromCLItem(ctx *context.ContentLibraryItemContextA2
 	vmi.Status.Name = clItem.Status.Name
 	vmi.Status.ProviderItemID = string(clItem.Spec.UUID)
 
-	return nil
+	return addContentLibraryRefToConversionAnnotation(ctx)
+}
+
+// addContentLibraryRefToConversionAnnotation adds the conversion annotation with the content
+// library ref value populated.
+func addContentLibraryRefToConversionAnnotation(ctx *context.ContentLibraryItemContextA2) error {
+	if ctx.CLItem.Status.ContentLibraryRef == nil {
+		return nil
+	}
+
+	current := ctx.VMI.DeepCopyObject().(conversion.Hub)
+
+	prevVMI := &v1alpha1.VirtualMachineImage{}
+	prev := prevVMI.DeepCopyObject().(conversion.Convertible)
+	if err := prev.ConvertFrom(current); err != nil {
+		return err
+	}
+
+	prevVMI = prev.(*v1alpha1.VirtualMachineImage)
+	prevVMI.Status.ContentLibraryRef = &corev1.TypedLocalObjectReference{
+		APIGroup: &imgregv1a1.GroupVersion.Group,
+		Kind:     ctx.CLItem.Status.ContentLibraryRef.Kind,
+		Name:     ctx.CLItem.Status.ContentLibraryRef.Name,
+	}
+
+	return utilconversion.MarshalData(prevVMI, ctx.VMI)
```

Comment:
Will all v1a2 VMI/CVMIs have the v1a1 VMI objs in its annotations ?. Just a little worried about the size of this annotation since the v1a1 VMIs have all those large annotations in it already.

Reply:
Yah I have the same concern. We really only need the `ContentLibraryRef` so can't we just stash that in the annotations, and deser that - if present - to populate the v1a1 field on down version conversion?

Reply:
In the `MarshalData` function, we delete the metadata from the source object, which in this case would be v1a1 VMI/CVMI. Only the spec and status of the object gets marshaled and stored into the annotation.

Reply:
The vmoperator.vmware.com/conversion annotation is just meant for the synthetic objects to just to handle old API client updates, not something stored in ectd. We don't need the whole v1a1 object here or do all of what the conversion hooks is already going to do. Things are a lot simpler and straightforward if we stash the ContentLibraryRef via the v1a2+ controllers in a dedicated annotation key so that the v1a1 ConvertFrom can use that (and if we don't want to have a gap from when upgrading to v1a1 to v1a2+ before the controller reconciles,  we can have the v1a1 ConvertTo stash that in the hub's annotation if not already set)